framework-cloud/framework-openshift/src/main/java/org/kie/cloud/openshift/scenario/GenericScenarioImpl.java

```diff
@@ -42,52 +41,38 @@
 
     private OpenShiftController openshiftController;
     private String projectName;
-    private WorkbenchDeploymentImpl workbenchDeployment;
-    private SmartRouterDeployment smartRouterDeployment;
-    private KieServerDeploymentImpl kieServerDeployment;
-    private DatabaseDeploymentImpl databaseDeployment;
 
-    private Map<String, String> envVariables;
-    private Map<ScenarioConstants, String> kieAppTemplates;
+    private List<WorkbenchDeployment> workbenchDeployments;
+    private List<KieServerDeployment> kieServerDeployments;
 
-    private static final Logger logger = LoggerFactory.getLogger(GenericScenarioImpl.class);
+    private List<DeploymentSettings> kieServerSettingsList;
+    private List<DeploymentSettings> workbenchSettingsList;
 
-    public GenericScenarioImpl(OpenShiftController openShiftController, Map<String, String> envVariables, Map<ScenarioConstants, String> kieAppTemplates) {
-        this.openshiftController = openShiftController;
-        this.envVariables = envVariables;
-        this.kieAppTemplates = kieAppTemplates;
-    }
+    private static final Logger logger = LoggerFactory.getLogger(GenericScenarioImpl.class);
 
-    @Override
-    public WorkbenchDeployment getWorkbenchDeployment() {
-        if (isWorkbenchInScenario()) {
-            return workbenchDeployment;
-        }
-        throw new RuntimeException("Workbench or Kie controller deployment is not in this scenario");
-    }
+    public GenericScenarioImpl(OpenShiftController openshiftController, List<DeploymentSettings> kieServerSettingsList, List<DeploymentSettings> workbenchSettingsList) {
+        this.openshiftController = openshiftController;
+        this.kieServerSettingsList = kieServerSettingsList;
+        this.workbenchSettingsList = workbenchSettingsList;
 
-    @Override
-    public SmartRouterDeployment getSmartRouterDeployment() {
-        if (isSmartRouterInScenario()) {
-            return smartRouterDeployment;
-        }
-        throw new RuntimeException("Smart router is not in this scenario");
+        workbenchDeployments = new ArrayList<>();
+        kieServerDeployments = new ArrayList<>();
     }
 
     @Override
-    public KieServerDeployment getKieServerDeployment() {
-        if (isKieServerInScenario()) {
-            return kieServerDeployment;
+    public List<WorkbenchDeployment> getWorkbenchDeployments() {
+        if (workbenchSettingsList.isEmpty()) {
+            throw new RuntimeException("Workbench or Kie controller deployment is not in this scenario");
```

Comment:
Why there is this check? If there aren't any deployments defined then we can return empty list, no need to throw the exception.